contrib/gcppubsub/pkg/adapter/adapter.go

```diff
@@ -89,40 +94,34 @@ func (a *Adapter) receiveMessage(ctx context.Context, m PubSubMessage) {
 	}
 }
 func (a *Adapter) postMessage(ctx context.Context, logger *zap.SugaredLogger, m PubSubMessage) error {
+	var err error
+	a.initClientOnce.Do(func() {
```

Comment:
why not do this in Start? I think it would be better to fail during start. But guess the fact that I'm asking this question here and in k8sevents means that there should probably be some comments added.

Reply:
whyyyy not. ok done.